README
Astro::Catalog module
---------------------

What is it?
-----------

The module is a generic object orientated astronomical catalogue object.

Requirements
------------

The madatory requirement of the Astro::Catalog package are,

  Astro::Coords
  Astro::Telescope
  Time::Piece
  File::Spec
  Math::Libm
  Net::Domain
  LWP::UserAgent
  SOAP::Lite
  Test
  Test::More
  Data::Dumper
  Carp
  Astro::VO::VOTable
  Astro::Flux
  Number::Uncertainty
  Misc::Quality

See http://heasarc.gsfc.nasa.gov/classx/votable/index.html to obtain the GSFC
Astro::VO::VOTable package. You need V0.9 or later of this module.

Provides
--------

The package provides the following user level classes

  Astro::Catalog
    A generic catalogue object, includes pluggable I/O support.

  Astro::Catalog::Item
    A generic catalogue item.

NB: Astro::Catalog::Star is maintained for backwards compatibility.

The following classes can be used to query various online databases
for astronomical objects. They all return an Astro::Catalog object
populated with matching Astro::Catalog::Star objects.

  Astro::Catalog::Query::2MASS
    Queries the Two Micron All Sky Survey (2MASS) catalogue.

  Astro::Catalog::Query::CMC
    Queries the Carlsberg Meridian Catalogue (CMC/12).

  Astro::Catalog::Query::GSC
    Queries the Guide Star Catalog (GSC).

  Astro::Catalog::Query::MPC
    Queries the Minor Planet Center database for solar system objects.

  Astro::Catalog::Query::Sesame
    Queries the CDS Sesame web service.

  Astro::Catalog::Query::SIMBAD
    Queries the CDS SIMBAD database. Implements most of the functionality
    of the Astro::SIMBAD module.

  Astro::Catalog::Query::SkyCat
    Queries SkyCat catalogues. For some catalogue queries, parsing of the
    returned catalogue will not work. In some cases the URLs of the
    resources may break this module's parsing code.

  Astro::Catalog::Query::SuperCOSMOS
    Queries the SuperCOSMOS catalogue.

  Astro::Catalog::Query::USNOA2
    Queries the USNO-A2.0 catalogue.

  Astro::Catalog::Query::Vizier
    Queries Vizier catalogues. For some catalogue queries, parsing of the
    returned catalogue will not work.

Additional classes provided by the package handle things that you don't
need to concern yourself with, unless you want to write your own I/O or
query plugins, these are

  Astro::Catalog::IO::Astrom     (write only)
  Astro::Catalog::IO::ASSM       (read only)
  Astro::Catalog::IO::Binary     (read only)
  Astro::Catalog::IO::Cluster
  Astro::Catalog::IO::FINDOFF
  Astro::Catalog::IO::FITSTable
  Astro::Catalog::IO::GaiaPick   (read only)
  Astro::Catalog::IO::JCMT
  Astro::Catalog::IO::LCOGFITSTable
  Astro::Catalog::IO::Northstar  (read only)
  Astro::Catalog::IO::RITMatch
  Astro::Catalog::IO::SExtractor (read only)
  Astro::Catalog::IO::STL        (read only)
  Astro::Catalog::IO::Simple
  Astro::Catalog::IO::TST        (read only)
  Astro::Catalog::IO::UKIRTBS    (read only)
  Astro::Catalog::IO::VOTable
  Astro::Catalog::IO::XY
  Astro::Catalog::Transport::REST
  Astro::Catalog::Transport::WebService
  Astro::Catalog::Query
  Astro::Catalog::Item::Morphology

The user level query classes inherit from the Astro::Catalog::Transport::XXX
classes, which in turn all inherit basic functionality from the Query base
class Astro::Catalog::Query. The Astro::Catalog::IO modules provide pluggable
polymorphic input/output.

It should be noted that Astro::Catalog::IO::TST, STL, GaiaPick, SExtractor,
and UKIRTBS do not currently provide a _write_catalog() function, and
Astrom does not currently provide a _read_catalog() function.

Where can I get it?
-------------------

The package is available from CPAN <http://www.cpan.org/> and from github
at https://github.com/timj/perl-Astro-Catalog

Installation
------------

Installation is automated using the Module::Build library

     % perl Build.PL
     % ./Build
     % ./Build test
     % ./Build install

Authors
-------

The module was originally written by Alasdair Allan <aa@astro.ex.ac.uk>,
of the University of Exeter Astrophysics Group (http://www.astro.ex.ac.uk/)
as part of his work for the eSTAR project.

The internal architecture of the module was extensively rewritten for v3.0
by Tim Jenness <t.jenness@jach.hawaii.edu> of the Joint Astronomy Centre
in Hawaii and Alasdair Allan <aa@astro.ex.ac.uk>, of the University of
Exeter Astrophysics Group. Then rewritten again by Alasdair Allan <aa@astro.ex.ac.uk> for the v4.0 release.

This version also includes patches supplied by Stephen Quinney
<S.J.Quinney@durham.ac.uk> of Durham University, Tim Lister
<tlister@lcogt.net> of Las Cumbres Observatory, and Luca Rizzi
<l.rizzi@jach.hawaii.edu> and Brad Cavanagh
<b.cavanagh@jach.hawaii.edu> of the Joint Astronomy Centre in Hawaii.

License
-------
This package is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License as published by the
Free Software Foundation; either version 2 of the License, or (at your
option) any later version.

This program is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
59 Temple Place,Suite 330, Boston, MA  02111-1307, USA


Changes

Release Notes
=============

For more detail please see the commit log at

  https://github.com/timj/perl-Astro-Catalog

v4.37

  + Tk::AstroCatalog window now resizable.
  + JCMT format updated with addition of proper motion and parallax.
  + More consistent spelling of catalog.

v4.36

  + Added IO modules for VEX, Hedwig and JCMT_OT_SC formats.
  + For Vizier query, generate combined c.rm parameter.
  + Use Astro::Catalog::Item internally instead of Astro::Catalog::Star.
  + Added catalog_convert script to convert catalogs between formats.

v4.35

  + Changed the Astro::Catalog::IO::TST pattern match used to identify
    magnitude error columns to allow multiple characters in the filter name
    (with thanks to Paul Leyland - CPAN bug #132433).
  + Suppressed some debugging output (unless $DEBUG is set).

v4.34

  + Updated reference coordinates for 4_sesame.t test.

v4.33

  + Updated the test scripts to work with the latest release
    of Astro::Coords (0.19).

v4.32

  + Updated skycat.cfg file.
  + Updated RITMatch for the case of multiple filters and renamed
    the write_catalog "waveband" option to "filter".
  + Adjusted JCMT catalog formatting to match the current catalog.
  + Skipped 2_jcmt.t's test of the HOLO source in the JCMT catalog.
  + Now skip test 4_skycat_gsc.t as service has been decommissioned.
  + Updated reference coordinates for 4_sesame.t test.

v4.31

  + Fix test 4_skycat_gsc.t for users who have a SkyCat configuration
    file which does not include "gsc@eso".

v4.30

  + Catalogs can now be sorted by distance in azimuth.

v4.28

  + Mercury has been added to the list of planets to be included
    in the JCMT catalog.

v4.27

  + Test scripts skip query tests in the event of network problems.

v4.26

  + The search button in Tk::AstroCatalog is now operational.
  + Minor updates to documentation.
  + Fixes for SkyCat GSC test due to hash key randomization.

v4.25

  + Tk::AstroCatalog can now be told to display additional columns
    in the widget.

v4.24

  + recommend that Astro::FITS::CFITSIO is installed rather than
    requiring it. We can't guarantee that people have the CFITSIO
    library installed.

v4.23

  + Tk::AstroCatalog now calls the onDestroy method correctly
    when the widget is closed from the window manager (Graham Bell)

v4.22

  + Use File::Temp for temporary files in tests.
  + Can now take IO::Handle objects for read_catalog and
    write_catalog.

v4.21

  + Remove code that relies on CVS $Revision$ tags. Those
    modules have now been updated to v2.00 to ensure that they
    are seen as new.

v4.2

  + The MPC web service seems to have changed so MPC is not functioning
    in this release.

  - Added FITSTable, ASSM, NorthStar, Binary, RITMatch, XY
    and LCOGFITSTable formats.
  - Now uses Module::Build
  - SuperCOSMOS and SIMBAD query classes have been removed since the
    remote format has changed.
  - Updated skycat.cfg file.
  - Astro::VO::VOTable is now an optional module.


Build.PL
#!perl

use strict;
use warnings;
use Module::Build;
use File::Spec;
use Config;

# Copyright (C) 2012-2013 Science and Technology Facilities Council.
# All Rights Reserved.

# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 3 of the License, or
# (at your option) any later version.

# This program is distributed in the hope that it will be useful,but
# WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# General Public License for more details.

# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place,Suite 330, Boston, MA 02111-1307,
# USA

# Set it up.
my $build = Module::Build->new(
    module_name => 'Astro::Catalog',
    dist_abstract => "A generic astronomical catalog class",
    license  => 'gpl2',
    author => [
        'Alasdair Allan <aa@astro.ex.ac.uk>',
        'Tim Jenness <tjenness@cpan.org>',
    ],
    dist_version => '4.36',
    meta_merge => {
        resources =>  {
            repository => "git://github.com/timj/perl-Astro-Catalog.git",
            homepage => "http://github.com/timj/perl-Astro-Catalog/tree/master",
        },
    },
    requires => {
        'LWP'                 => 5.53,
        'Net::Domain'         => 2.13,
        'Pod::Usage'          => 1.14,
        'Astro::Coords'       => 0.12,
        'File::Spec'          => 0.84,
        'Astro::Telescope'    => 0.50,
        'Time::Piece'         => 1.08,
        'SOAP::Lite'          => 0.00,
        'Carp'                => 1.01,
        'Data::Dumper'        => 2.12,
        'Scalar::Util'        => 1.14,
        'File::Temp'          => 0.14,
        'Class::Struct'       => 0.63,
        'Number::Uncertainty' => 0.01,
        'Misc::Quality'       => 0.01,
        'DateTime'            => 0,
        'DateTime::Format::ISO8601' => 0,
        'Astro::Flux'         => 0.01
    },
    recommends => {
        'Astro::FITS::CFITSIO'=> 1.03,
        'Astro::VO::VOTable' => 0.90,
        'Astro::VEX' => 0,
        'Text::CSV' => 0,
        'Tk' => 804.029,
    },
    configure_requires => {
        "Module::Build" => 0.30,
    },
    build_requires => {
        'Test::More' => 0,
    },
    script_files => [
        'bin/catalog_convert',
    ],
    etc_files => {
        'etc/skycat.cfg' => 'etc/skycat.cfg',
        'etc/sss.cfg' => 'etc/sss.cfg',
    },
    install_path => {
        'etc' => File::Spec->catdir($Config{installprefix}, 'etc'),
    },
);

$build->add_build_element('etc');
$build->create_build_script;
